### PR TITLE
testing: add `assert_no_diff` helper to assist pytest users

### PR DESCRIPTION
**Pull request by joesolly**

### CODE REVIEW

**Comment by joesolly** [comment]
@jirikuncar thanks for merging my other PR and being an active maintainer. Do you have thoughts on this PR? I wind up implementing this code in almost every repo I work on.

File: dictdiffer/__init__.py [hunk 1]
```diff
@@ -388,3 +390,27 @@ def revert(diff_result, destination, in_place=False):
                      and return the destination structure.
     """
     return patch(swap(diff_result), destination, in_place)
+
+
+def assert_no_diff(*args, **kwargs):
```
Comment (jirikuncar) [inline dictdiffer/__init__.py:395]: I would see this function in `utils`.

Comment (joesolly) [reply dictdiffer/__init__.py]: adding to `utils` creates a circular import, I added it to a new file testing_utils.py instead.

File: tests/test_testing_utils.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# This file is part of Dictdiffer.
+#
+# Copyright (C) 2015 CERN.
```
Comment (jirikuncar) [inline tests/test_testing_utils.py:3]:
```suggestion
# Copyright (C) 2021 CERN.
```

File: dictdiffer/testing_utils.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# This file is part of Dictdiffer.
+#
+# Copyright (C) 2015 CERN.
+# Copyright (C) 2017, 2019 ETH Zurich, Swiss Data Science Center, Jiri Kuncar.
```
Comment (jirikuncar) [inline dictdiffer/testing_utils.py:4]:
```suggestion
# Copyright (C) 2021 CERN.
```

File: dictdiffer/testing_utils.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+#
+# Dictdiffer is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more
+# details.
+"""Testing Utils gathers helper functions for testing."""
```
Comment (jirikuncar) [inline dictdiffer/testing_utils.py:9]:
```suggestion
"""Define helper functions for testing."""
```

File: dictdiffer/testing_utils.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# This file is part of Dictdiffer.
```
Comment (jirikuncar) [inline dictdiffer/testing_utils.py:1]:
```console
$ git mv dictdiffer/testing_utils.py dictdiffer/testing.py
```

File: tests/test_testing_utils.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+#
+# Dictdiffer is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+import unittest
+
+import pytest
+
+from dictdiffer.testing_utils import assert_no_diff
```
Comment (jirikuncar) [inline tests/test_testing_utils.py:13]:
```suggestion
from dictdiffer.testing import assert_no_diff
```

File: tests/test_testing_utils.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# This file is part of Dictdiffer.
```
Comment (jirikuncar) [inline tests/test_testing_utils.py:1]:
```console
$ git mv tests/test_testing_utils.py test/test_testing.py
```

**Review by jirikuncar — COMMENTED** [review]
There are few minor issues and then it can be merged. Thanks a lot!

**Comment by joesolly** [comment]
> There are few minor issues and then it can be merged. Thanks a lot!

@jirikuncar Thanks for all the feedback, I think I addressed it all.

**Review by jirikuncar — APPROVED** [review]

**Pull request merged**